src/HttpClient/HttpClientFactory.php

```diff
@@ -121,9 +121,15 @@ public function create(Options $options): HttpAsyncClientInterface
      */
     private function resolveClient(Options $options)
     {
+        $httpTimeout = $options->getHttpTimeout() ?? self::DEFAULT_HTTP_TIMEOUT;
+        $httpConnectTimeout = $options->getHttpConnectTimeout() ?? self::DEFAULT_HTTP_CONNECT_TIMEOUT;
+
         if (class_exists(SymfonyHttplugClient::class)) {
             $symfonyConfig = [
-                'max_duration' => self::DEFAULT_HTTP_TIMEOUT,
+                /*
+                 * {@see https://symfony.com/doc/current/http_client.html#dealing-with-network-timeouts}
+                 */
+                'max_duration' => $httpConnectTimeout + $httpTimeout,
```

Comment:
@ste93cry do you agree with this change?

Reply:
We're changing a default here, so technically this is a breaking change. Other than that, the value is almost doubled so while I agree that if the underlying lib is not supporting splitted options this is the only way, I would leave this as it was before

Reply:
Ok so we need this:
```suggestion
                'max_duration' => $httpTimeout,
```

Reply:
I think it would be a good idea to set the SymfonyHttplugClient `timeout` option to `$httpConnectTimeout` here.

It's a bit confusing, because the Symfony HttpClient `timeout` option applies to any idle time, whether while connecting or while waiting for the response.  But despite the semantic difference, it's useful because setting this to e.g. the 2-second default for connect timeout allows the HttpClient to give up more quickly if there is no connection after 2 seconds, while still having 5 seconds max-duration configured for the full request and response.

Adding the `timeout` setting does cause an additional timeout to apply if there is a 2-second pause in data arriving during the response phase, but that seems fine to me and totally compatible with the default 5-second max duration (if you have huge network latency, then increasing the HttpConnectTimeout would give you more time for both connecting and waiting for the response).

In my testing w/ Sentry SDK + Symfony HttpClient, setting `timeout` to 2 seconds allowed Sentry SDK to give up several seconds more quickly when trying to connect to an unreachable Sentry server (and no side effects with working Sentry server).

Reply:
Thanks for digging through this @mfb! So your suggestion is:
```suggestion
                'timeout' => $httpConnectTimeout,
```

Reply:
I was testing this actually:
```
                    'max_duration' => self::DEFAULT_HTTP_TIMEOUT,
                    'timeout' => self::DEFAULT_HTTP_CONNECT_TIMEOUT,
```
I think it makes sense to have 5 second end-to-end max duration, as we do now, and within that, a 2 second timeout for idle/connect time.  It seems like the two settings are less useful and more confusing if they get added together.

Reply:
By the way I was playing with very low values of max_duration and timeout, and noticed that you can make the SDK send the event successfully but also give up before receiving the response and believe the send failed, which is actually great from a performance perspective; of course this would also increase the rate of actual failures assuming you have sporadic latency :)

Reply:
By the way the symfony docs say `Passing 0 as timeout allows monitoring responses in a non-blocking way.`
Or is there another way to configure the sentry SDK asynchronously?

Reply:
>I think it would be a good idea to set the SymfonyHttpClient timeout option to $httpConnectTimeout here.

I updated the PR accordingly 👍

>Or is there another way to configure the sentry SDK asynchronously?

I think that this applies if you use something along the lines of `$client->stream()` to stream the responses, which is not the case with the transport we provide out-of-the-box